apps/aeutils/src/aeu_requests.erl

```diff
@@ -61,21 +61,36 @@ block(Peer, Hash) ->
 -spec transactions(aec_peers:peer()) -> response([aec_tx:signed_tx()]).
 transactions(Peer) ->
     Uri = aec_peers:uri(Peer),
-    Response = process_request(Uri, post, "transactions", []),
-    case Response of
-        {ok, Map} ->
-            Txs = maps:get('Transactions', Map, []),
-            {ok, lists:map(
-                   fun(T) ->
-                           aec_tx_sign:deserialize_from_binary(
-                             base64:decode(T))
-                   end, Txs)}
+    Response = process_request(Uri, get, "transactions"),
+    lager:debug("transactions Response = ~p", [Response]),
+    try Txs = tx_response(Response),
+         try {ok, lists:map(
+                    fun(#{<<"tx">> := T}) ->
+                            aec_tx_sign:deserialize_from_binary(
+                              base64:decode(T))
+                    end, Txs)}
+         catch
+             error:Reason ->
+                 lager:error("Error decoding transactions: ~p", [Reason]),
+                 {error, Reason}
+         end
+    catch
+        error:_Reason1 ->
+            lager:error("Wrong response type: ~p", [Response]),
+            {error, wrong_response_type}
     end.
 
+tx_response({ok, #{'Transactions' := Txs}}) -> Txs;
```

Comment:
why do we have two different keys in the map envelope

Reply:
It follows from the Swagger specification: `'Transactions'` is an array of transaction objects; `<<"tx">>` is an envelope around the serialized tx binary.

Unfortunately, swagger seems to have problems with more complex array definitions (e.g. 'an array of objects that are themselves arrays of objects'). If someone can achieve a working solution for that, it would be great.